### PR TITLE
Support for scaling ReplicaSets

### PR DESCRIPTION
**Pull request by matkam**
👋 This PR allows for scaling on ReplicaSets rather than just Deployments. I did some basic testing on a Kubernetes cluster to make sure it works as intended. Please let me know if you'd like any more changes before merging.

### CODE REVIEW

File: pkg/controller/controller.go [hunk 1]
```diff
@@ -94,8 +94,10 @@ type Controller struct {
 	kubeclientset kubernetes.Interface
 	// customclientset is a clientset for our own API group
 	customclientset            clientset.Interface
-	deploymentsLister          appslisters.DeploymentLister
+	deploymentLister           appslisters.DeploymentLister
 	deploymentsSynced          cache.InformerSynced
+	replicaSetLister           appslisters.ReplicaSetLister
+	replicaSetsSynced          cache.InformerSynced
```
Comment (alok87) [inline pkg/controller/controller.go:100]: `deploymentLister`(singular) and `replicaSetsLister` (plural). Is it intentional?

File: README.md [hunk 2]
```diff
@@ -81,12 +81,15 @@ Beanstalk's queueURI would be like: `beanstalk://beanstalkDNSName:11300/test-tub
...
 | targetMessagesPerWorker |  Number of jobs in the queue which have not been picked up by the workers. This is also used to calculate the desired number of workers. | Yes |
 | secondsToProcessOneJob | This metric is useful to calculate the desired number of workers more accurately. It is particularly very useful for workers which have `targetMessagesPerWorker` as always zero. `secondsToProcessOneJob` in the combination with `messagesSentPerMinute`(queue RPM) helps in calculating the minimum workers that is expected to be running to handle `messagesSentPerMinute`(RPM) with every job being processed in `secondsToProcessOneJob` seconds. (highly recommended, default=0.0 i.e. disabled). | No |
 | maxDisruption | Amount of disruption that can be tolerated in a single scale down activity. Number of pods or percentage of pods that can scale down in a single down scale down activity. Using this you can control how fast a scale down can happen. This can be expressed both as an absolute value and a percentage. (default is the WPA flag `--wpa-default-max-disruption`). | No |
 
+*It is mandatory to set either `deploymentName` or `replicaSetName`.
```
Comment (alok87) [inline README.md:91]: possible to put this in CRD validation?

**Comment by alok87** [comment]
Thanks for the PR :)

Curious! what is your use case? Why are you using replicaSet when we have deployments? 
Please rebase with master and fix the travis.

**Comment by matkam** [comment]
Thanks for writing this autoscaler!
The use case of this change is compatibility with our deployment operator, [Picchu](https://github.com/Medium/picchu/). We wanted more control over the release of new revisions, rather than relying on Deployment magic, and opted to have Picchu manipulate ReplicaSets instead of Deployments.

File: README.md (hunk 2 above)
Comment (matkam) [reply README.md:91]: I made the CRD change to include `oneOf`, and it looks like it works exactly the way it should! It fails validation if neither are defined:
```
"spec" must validate one and only one schema (oneOf). Found none valid
spec.deploymentName in body is required
```
And when more than one is defined:
```
"spec" must validate one and only one schema (oneOf). Found 2 valid alternatives
```

**Comment by alok87** [comment]
We have not released it in Github release yet. But have pushed the following docker images for use:
```
pushed: practodev/workerpodautoscaler:v1.0.0-19-g59d0d48
pushed: practodev/workerpodautoscaler:v1.0
pushed: practodev/workerpodautoscaler:v1
```

**Pull request merged**